pilot/pkg/bootstrap/server.go

```diff
@@ -1183,7 +1185,56 @@ func (s *Server) initMeshHandlers() {
 	})
 }
 
-func (s *Server) initWorkloadTrustBundle() {
+func (s *Server) addIstioCAToTrustBundle(args *PilotArgs) {
+	var err error
+
+	if s.CA != nil {
+		// When CA is enabled, fetch TrustBundle from CA
+		rootCerts := []string{string(s.CA.GetCAKeyCertBundle().GetRootCertPem())}
+		err = s.workloadTrustBundle.UpdateTrustAnchor(&tb.TrustAnchorUpdate{
+			TrustAnchorConfig: tb.TrustAnchorConfig{Certs: rootCerts},
+			Source:            tb.SourceIstioCA,
+		})
+		if err != nil {
+			log.Errorf("fatal: unable to add CA root as trustAnchor")
+		}
+	} else if s.kubeClient != nil {
+		// When CA is not enabled, fetch root-cert.pem from CA secrets in cluster
```

Comment:
I am about concerned about reading `cacerts`. This is not currently being done, AFAIK - its mounted as a file. I don't know if this will cause problems but it seems like it might

Reply:
I thought I had covered all possible scenarios here, but agree with you. Better to read this from file-system, similar to CA logic.